src/muxer.rs

```diff
@@ -464,20 +464,20 @@ fn offset<'a>(original: &(&'a [u8], usize), subslice: &(&'a [u8], usize)) -> usi
     second + subslice.1 - first - original.1
 }
 
-pub fn gen_mkv_prefix<'a>(
+#[allow(dead_code)]
+fn gen_mkv_prefix<'a>(
     input: (&'a mut [u8], usize),
-    header: &EBMLHeader,
-    seek_head: &SeekHead,
-    info: &Info,
-    tracks: &Tracks,
+    header: &'a EBMLHeader,
+    seek_head: &'a SeekHead,
+    info: &'a Info,
+    tracks: &'a Tracks,
```

Comment:
This seems wrong. The lifetime of e.g. `header` has nothing to do with the lifetime of the returned `&mut [u8]`.

Reply:
I can remove the lifetime neither from `gen_mkv_prefix` nor from the other functions (like gen_ebml_header for example)

Reply:
See https://github.com/rust-av/matroska/pull/79#discussion_r823688286

Reply:
I changed the lifetime as suggested in [#79 (comment)](https://github.com/rust-av/matroska/pull/79#discussion_r823688286) but it doesn't seem to solve the problem. A specific lifetime is necessary because `h.doc_type` contains the value extracted from the slice